mail/views.py

```diff
@@ -64,11 +67,47 @@ def post(self, request, *args, **kargs):  # pylint: disable=unused-argument, no-
         )
 
 
+class CourseTeamMailView(GenericAPIView):
+    """
+    View class that handles HTTP requests to course team mail API
+    """
+    serializer_class = GenericMailSerializer
+    authentication_classes = (
+        authentication.SessionAuthentication,
+        authentication.TokenAuthentication,
+    )
+    permission_classes = (permissions.IsAuthenticated, )
+    lookup_field = "id"
+    lookup_url_kwarg = "course_id"
+    queryset = Course.objects.all()
+
+    def post(self, request, *args, **kargs):  # pylint: disable=unused-argument, no-self-use
+        """
+        POST method handler
+        """
+        user = request.user
+        course = self.get_object()
+        if not ProgramEnrollment.objects.filter(user=user, program=course.program).exists():
+            raise ValidationError('The user sending the course team message must be enrolled in the given program.')
```

Comment:
A couple things:

1. You could only spam a course team email using this view, so at the very least it reduces the risk.
1. It shouldn't be too much trouble to add the restriction on the API side. I'll work on that